apps/pygame/src/tangl/pygame_client/__main__.py

```
@@ -93,11 +217,30 @@ def main(argv: list[str] | None = None) -> int:
+        # flag has no way to supply, and sending its bare activation payload
+        # reaches the backend as a malformed move rather than as a skipped turn.
+        commits = [
+            action
+            for choice in frame.choices
+            if isinstance(action := choice_action(choice), Commit)
+        ]
+        if not commits:
+            blocked = [
+                choice.text
+                for choice in frame.choices
+                if choice.available and choice_action(choice) is None
```

Ruff (0.16.3)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)